testing/matrix/linter/rules/resources/vpa.go

```diff
@@ -146,6 +210,62 @@ func parseVPATargetIndex(scope *lintingScope, vpaObject storage.StoreObject) (st
 	return target, true
 }
 
+// ensureVPAContainersMatchControllerContainers verifies VPA container names in resourcePolicy match corresponding controller container names
+func ensureVPAContainersMatchControllerContainers(scope *lintingScope, object storage.StoreObject, index storage.ResourceIndex, vpaContainerNamesMap map[storage.ResourceIndex]map[string]struct{}) bool {
+	vpaContainerNames, ok := vpaContainerNamesMap[index]
```

Comment:
I see that sets would be handy here as well. But I won't insist on refactoring. As for me, I would do refactor from `map[storage.ResourceIndex]map[string]struct{}` to `map[storage.ResourceIndex]set.Set`.